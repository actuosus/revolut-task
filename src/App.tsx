import React from "react";
import { NavigationScreenProp, NavigationState } from "react-navigation";
import { Provider } from "react-redux";
import RX from "reactxp";
import AppContainer from "./containers/App";
import AppNavigator from "./containers/Navigator";
import configureStore from "./store";
import initialState from "./store/initial";

if (typeof window === "undefined") {
  global.window = {
    innerWidth: 375,
    innerHeight: 600
  };
}

const { store } = configureStore(initialState);

interface AppProps {
  navigation?: NavigationScreenProp<NavigationState>;
}

export class App extends RX.Component<AppProps> {
  public static router = AppNavigator.router;

  public render() {
    const { navigation } = this.props;

    return (
      <Provider store={store}>
        <AppContainer navigation={navigation} />
      </Provider>
    );
  }
}

export default App;
