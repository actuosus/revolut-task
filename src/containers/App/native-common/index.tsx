import { createAppContainer } from "react-navigation";
import AppNavigator from "../../Navigator";

export const AppContainer = createAppContainer(AppNavigator);

export default AppContainer;
