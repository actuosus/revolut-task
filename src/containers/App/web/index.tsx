import { createBrowserApp } from "@react-navigation/web";
import RX from "reactxp";
import AppNavigator from "../../Navigator";

let AppContainer: React.ElementType;

if (process.browser) {
  AppContainer = createBrowserApp(AppNavigator);
  // AppContainer = WebAppContainer;
} else {
  // @ts-ignore
  RX.ActivityIndicator._isStyleSheetInstalled = true;
  AppContainer = AppNavigator;
}

export default AppContainer;
