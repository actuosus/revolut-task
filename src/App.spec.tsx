import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('<App />', () => {
  it('matches its snapshot', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
