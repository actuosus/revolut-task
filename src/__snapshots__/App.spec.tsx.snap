// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches its snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "liftedStore": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Navigator />
</Provider>
`;
